### PR TITLE
feat: add message to status change for negotiations

### PR DESCRIPTION
**Pull request by svituz**
### Description:

This PR adds the possibility of sending a message when a status change for the negotiation is sent.
When the message is sent, it is added as a Public Post to the Negotiation.
The message is also added as part of the notification email.
Also, the list of possible events for negotiation is changed, since now it returns all the fields of the events 

### Checklist:

_Make sure you tick all the boxes below if they are true or do not apply before you ask for review_

- [x] I have performed a self-review of my code
- [x] I have made my code as simple as possible
- [x] I have added relevant tests for my changes and the code coverage has not dropped substantially
- [x] I have removed all commented code
- [x] I have updated the documentation in all relevant places (Javadoc, Swagger, MDs...)
- [x] I have described the PR and added a meaningful title in the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationController.java [hunk 1]
```diff
@@ -162,17 +164,28 @@ public EntityModel<NegotiationDTO> retrieve(@Valid @PathVariable String id) {
...
+      @RequestBody(required = false) @Nullable
+          NegotiationUpdateLifecycleDTO negotiationUpdateLifecycleDTO) {
     if (!AuthenticatedUserContext.isCurrentlyAuthenticatedUserAdmin()
         && !isCreator(negotiationService.findById(id, false))) {
       throw new ResponseStatusException(HttpStatus.FORBIDDEN);
     }
     // Process the request
-    negotiationLifecycleService.sendEvent(id, event);
+    String message = null;
+    if (negotiationUpdateLifecycleDTO != null
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationController.java:184]: Objects.isNull()

File: src/main/java/eu/bbmri_eric/negotiator/negotiation/state_machine/negotiation/NegotiationLifecycleService.java [hunk 2]
```diff
@@ -25,6 +25,7 @@ public interface NegotiationLifecycleService {
    *
    * @return the new status of the Negotiation
    */
-  NegotiationState sendEvent(String negotiationId, NegotiationEvent negotiationEvent)
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/negotiation/state_machine/negotiation/NegotiationLifecycleService.java:28]: leave the orgiginal signature, just add a new method with update signature. Then the parameter will indeed be optional

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
please see comments plus I'm still not convinced we need the required parameter. It could just awayls be optional and the code would be much simpler. PLus the backend must send an email where this comment is shown. PLease see the original issue

File: src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationController.java (hunk 1 above)
Comment (svituz) [reply src/main/java/eu/bbmri_eric/negotiator/negotiation/NegotiationController.java:184]: The documentation says:
This method exists to be used as a [Predicate](https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html), filter(Objects::isNull)

**Comment by svituz** [comment]
> please see comments plus I'm still not convinced we need the required parameter. It could just awayls be optional and the code would be much simpler. PLus the backend must send an email where this comment is shown. PLease see the original issue

My understanding was that, when declining a negotiation, the administrator must specify the reason for rejection. So I came up with the messageRequired flag.

**Review by RadovanTomik — APPROVED** [review]

**Pull request merged**